benchmarks/ckks_basic.cpp

```diff
@@ -105,67 +225,80 @@ static void multiply_and_add_two_ciphertexts(benchmark::State& state,
   meta.data->publicKey.Encrypt(ctxt2, ptxt2);
   // Benchmark multiplying and adding ciphertexts
   for (auto _ : state) {
-    ctxt1.multiplyBy(ctxt2);
+    state.PauseTiming();
+    auto copy(ctxt1);
+
+    state.ResumeTiming();
+    copy.multiplyBy(ctxt2);
     benchmark::DoNotOptimize(ctxt1 += ctxt2);
   }
-  std::cout << "Operations performed = " << state.iterations() << std::endl;
 }
 
 Meta fn;
-Params tiny_params(1024, 1, 5800);
-BENCHMARK_CAPTURE(adding_two_ciphertexts, tiny_params, fn(tiny_params))
-    ->Unit(benchmark::kMillisecond)
-    ->Iterations(200)
-    ->Complexity();
-BENCHMARK_CAPTURE(multiplying_two_ciphertexts, tiny_params, fn(tiny_params))
-    ->Unit(benchmark::kMillisecond)
-    ->Iterations(200);
-BENCHMARK_CAPTURE(encrypting_ciphertexts, tiny_params, fn(tiny_params))
-    ->Unit(benchmark::kMillisecond)
-    ->Iterations(200);
-BENCHMARK_CAPTURE(decrypting_ciphertexts, tiny_params, fn(tiny_params))
-    ->Unit(benchmark::kMillisecond)
-    ->Iterations(200);
-BENCHMARK_CAPTURE(multiply_and_add_two_ciphertexts,
-                  tiny_params,
-                  fn(tiny_params))
-    ->Unit(benchmark::kMillisecond)
-    ->Iterations(200);
-
-Params small_params(16384, 1, 5800);
-BENCHMARK_CAPTURE(adding_two_ciphertexts, small_params, fn(small_params))
-    ->Unit(benchmark::kMillisecond)
-    ->MinTime(200);
-BENCHMARK_CAPTURE(multiplying_two_ciphertexts, small_params, fn(small_params))
-    ->Unit(benchmark::kMillisecond)
-    ->MinTime(200);
-BENCHMARK_CAPTURE(encrypting_ciphertexts, small_params, fn(small_params))
-    ->Unit(benchmark::kMillisecond)
-    ->Iterations(200);
-BENCHMARK_CAPTURE(decrypting_ciphertexts, small_params, fn(small_params))
-    ->Unit(benchmark::kMillisecond)
-    ->Iterations(200);
-BENCHMARK_CAPTURE(multiply_and_add_two_ciphertexts,
-                  small_params,
-                  fn(small_params))
-    ->Unit(benchmark::kMillisecond)
-    ->Iterations(200);
-
-Params big_params(65536, 1, 5800);
-BENCHMARK_CAPTURE(adding_two_ciphertexts, big_params, fn(big_params))
-    ->Unit(benchmark::kMillisecond)
-    ->MinTime(200);
-BENCHMARK_CAPTURE(multiplying_two_ciphertexts, big_params, fn(big_params))
-    ->Unit(benchmark::kMillisecond)
-    ->MinTime(200);
-BENCHMARK_CAPTURE(encrypting_ciphertexts, big_params, fn(big_params))
-    ->Unit(benchmark::kMillisecond)
-    ->Iterations(200);
-BENCHMARK_CAPTURE(decrypting_ciphertexts, big_params, fn(big_params))
-    ->Unit(benchmark::kMillisecond)
-    ->Iterations(200);
-BENCHMARK_CAPTURE(multiply_and_add_two_ciphertexts, big_params, fn(big_params))
-    ->Unit(benchmark::kMillisecond)
-    ->Iterations(200);
+Params tiny_params(1024, 1, 360);
```

Comment:
What are the params? m, r, qbits? Is `r`/`precision` really 1 here?

Reply:
I've added the names of the input params. They are `m`, `precision`, and `qbits`. Not sure we really want `precision=1` or what effect it has on the benchmark times.